javascript/javascript3/week2/currencyCalculator.js

```diff
@@ -0,0 +1,151 @@
+// select#currency-from.drop-list and select#currency-to.drop-list
+const currencyFromInput = document.querySelector("#currency-from");
+const currencyToInput = document.querySelector("#currency-to");
+
+// When Convert button is clicked, calculate exchange rate.
+const convertBtn = document.querySelector("#convert")
+
+// Display Variables
+const selectCountryFrom = document.querySelector("#currency-from")
+const selectCountryTo = document.querySelector("#currency-to")
+const displayCurrencyFrom = document.querySelector(".display-currency-from");
+const displayCurrencyTo = document.querySelector(".display-currency-to");
+const displayConversionRate = document.querySelector(".display-conversion-rate");
+const displayConvertedAmount = document.querySelector("#amount-to");
+const displayDateAndTime = document.querySelector(".display-date-time")
+
+//------------------------------------------------//
+
+// STEP 4: GET EXCHANGEINFO SAVED IN LOCAL STORAGE
+// Get exchangeInfo saved in Local Storage
+const exchangeInfoSaved = JSON.parse(localStorage.getItem("exchangeInfo"));
+
+if (exchangeInfoSaved) {
+    const countryCode = (exchangeInfoSaved["countryCode"])
+
+    for (let i = 0; i < countryCode.length; i++) {
+        // Create and append options for Country From
+        selectCountryFrom.innerHTML += `<option value="${countryCode[i]}">${countryCode[i]}</option>`
+        // Create and append options for Country To
+        selectCountryTo.innerHTML += `<option value="${countryCode[i]}">${countryCode[i]}</option>`
+    }
+}
+
+//-------------------------------------------------//
+
+// STEP 3: GET INPUT DATA FROM LOCAL STORAGE
+// Get Input Data saved in Local Storage
+const inputSaved = JSON.parse(localStorage.getItem("input"));
+
+// Check if input is saved in local storage.
+// If yes - display it as default.
+if (inputSaved) {
+    currencyFromInput.value = inputSaved["country-from"];
+    currencyToInput.value = inputSaved["country-to"];    
+    displayCurrencyFrom.innerHTML = `1 ${currencyFromInput.value} equals`;
+}
+//-------------------------------------------------//
+// STEP 1: FETCH DATA FROM API
+//Fetch Data from API
+const exchangeRatesData = async () => {
+
+    // When the convert button is clicked, calculate currency exchange.
+    let baseCurrency = currencyFromInput.value;
+   
+    // If location is blank, then default currency = "EUR"
+    if (baseCurrency == "") {
+        console.log("True");
+        baseCurrency = "EUR"
+    } else {
+        baseCurrency = currencyFromInput.value;        
+    }
+    
+    try {
+        const response = await fetch(`${BASE_URL}/v6/${API_KEY}/latest/${baseCurrency}`)
+        console.log(response.status);  //200 (Should be between 200 and 299)
+
+        const data = await response.json();
+        return data;
+    } catch (error) {
+        console.log(error)
+    }
+
+};
+
+// STEP 2: SAVE DATA TO LOCAL STORAGE
+const saveToLocalStorage = async () => {
```

Comment:
I would suggest just making this function take data as an input and saves the relevant data. That way, this function doesn't have to be asynchronous.